application_type_handlers_test.go

```diff
@@ -362,6 +388,39 @@ func TestApplicationTypeGet(t *testing.T) {
 	}
 }
 
+// TestApplicationTypeGetWithTenant tests that application type is returned
+// even when tenant is provided (the request usually doesn't need a tenant)
+func TestApplicationTypeGetWithTenant(t *testing.T) {
+	c, rec := request.CreateTestContext(
+		http.MethodGet,
+		"/api/sources/v3.1/application_types/1",
+		nil,
+		nil,
+	)
+
+	c.SetParamNames("id")
+	c.SetParamValues("1")
+
+	err := ApplicationTypeGet(c)
+	if err != nil {
+		t.Error(err)
+	}
+
+	if rec.Code != 200 {
+		t.Error("Did not return 200")
+	}
+
+	var outAppType m.ApplicationTypeResponse
+	err = json.Unmarshal(rec.Body.Bytes(), &outAppType)
+	if err != nil {
+		t.Error("Failed unmarshaling output")
+	}
+
+	if outAppType.DisplayName != "test app type" {
+		t.Error("ghosts infected the return")
+	}
+}
+
```

Comment:
Here the function description says that an app type is returned "even if the tenant is provided", but you don't set a tenant in the test, right?

Reply:
😂 good eye ... the tenant is very important here ... ha ha